### DISCUSSION TITLE
getContactPoints Clarification

### DISCUSSION THREAD

**Discussion by luarss** [bulletphysics/bullet3#3838] [Q&A] [answered]
I was recently implementing a code from https://github.com/facebookresearch/tacto

When running one of their examples (demo_pybullet_digit.py) and I tried printing out the contact points via p.getContactPoints, I only found that they returned 4 points. This is despite increasing the size of the object and force on which the object contact the other object. 

Is this by design? Any clarification would be appreciated.

**Comment by erwincoumans** [comment, marked as answer]
Yes, between two convex objects up to 4 points are returned (contact caching and contact reduction to 4 points is applied). For concave objects, such as a static (non-moving, mass=0) triangle mesh or a convex decomposition (for example generated using v-hacd) you may get more contact points.
